packages/web3-core-promievent/index.d.ts

```diff
@@ -0,0 +1,23 @@
+/*
+    This file is part of web3.js.
+    web3.js is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    web3.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+    You should have received a copy of the GNU Lesser General Public License
+    along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file index.d.ts
+ * @author Josh Stevens <joshstevens19@hotmail.co.uk>
+ * @date 2018
+ */
+
+export declare class PromiEvent {
+    proxyHandler(target: PromiEvent, name: string): Function;
+ }
+ 
```

Comment:
This type is not correct because I use the JS es6 proxy to combine all methods of the EventEmitter and Promise. The proxy handler itself can't be called because I'm returning the proxy object in the constructor and the ES6 proxy delegates the method calls.

Reply:
ahh ok so we probably don't need types for this then?

Reply:
We definitely need types for these :)